zerver/tests/tests.py

```diff
@@ -587,6 +590,17 @@ def test_add_bot(self):
         bot = bots[0]
         self.assertEqual(bot['bot_owner'], 'hamlet@zulip.com')
 
+    def test_add_bot_with_user_avatar(self):
+        # type: () -> None
+        self.login("hamlet@zulip.com")
+        self.assert_num_bots_equal(0)
+        with open(os.path.join(TEST_AVATAR_DIR, 'img.png'), 'rb') as fp:
+            self.create_bot(file=fp)
+        self.assert_num_bots_equal(1)
+
+        profile = get_user_profile_by_email('hambot-bot@zulip.com')
+        self.assertEqual(profile.avatar_source, UserProfile.AVATAR_FROM_USER)
```

Comment:
It's a little less than trivial to do this, since the file name from the upload gets thrown away, and we only store the bits.  Figuring out which bits got stored requires a get that follows a redirect.   Perhaps instead of letting this test get too far in the weeds on S3/local-storage implementation details, I could just mock upload_avatar_image() and be a little more thorough?